File: 은빈/자바/[PGS]최빈값구하기.java
import java.util.*;

class Solution {
    public int solution(int[] array) {
        int answer = 0;
        int maxCnt = 0;
        
        Map<Integer, Integer> map = new HashMap<>();
        
        for(int num : array) {
            int cnt = map.getOrDefault(num, 0) + 1;
            
            if(cnt > maxCnt) {
                maxCnt = cnt;
                answer = num;
            }
            else if(cnt == maxCnt) {
                answer = -1;
            }
            
            map.put(num, cnt);
        }
        
        return answer;
    }
}

File: 은빈/파이썬/[PGS]최빈값구하기.py
from collections import Counter

def solution(array):
    answer = -1
    cnt = [-1]
    
    counter = Counter(array)
    numSet = set(array)
    
    for num in numSet:
        myCnt = counter[num]
        
        if cnt[0] == myCnt:
            answer = num
            cnt.append(myCnt)
        elif cnt[0] < myCnt:
            answer = num
            cnt = [myCnt]
    
    if(len(cnt) > 1):
        answer = -1
    
    return answer